sdk/component/component.go

```diff
@@ -0,0 +1,71 @@
+package component
+
+import (
+	"context"
+
+	ocsf "github.com/ocurity/dracon/sdk/component/gen/com/github/ocsf/ocsf_schema/v1"
+)
+
+// Helpers interfaces for common functionalities.
+type (
+	// Validator allows validating findings by a specified criteria.
+	Validator interface {
+		// Validate validates the supplied findings and returns an error if invalid.
+		Validate(ctx context.Context, findings []*ocsf.VulnerabilityFinding) error
+	}
+
+	// Reader allows reading findings from a source.
+	Reader interface {
+		// Read reads and parses findings from a source.
+		Read(ctx context.Context) ([]*ocsf.VulnerabilityFinding, error)
+	}
+
+	// Filterer allows filtering out findings by a specified criteria.
+	Filterer interface {
+		// Filter returns filtered findings from the supplied ones.
+		Filter(ctx context.Context, findings []*ocsf.VulnerabilityFinding) ([]*ocsf.VulnerabilityFinding, error)
+	}
+)
+
+// Components interfaces.
+type (
```

Comment:
❓ Should we just merge the concepts of enrichers and producers?

I think that they do very similar things and they only differ in semantics:
- a producer writes vs an enricher updates
- an enricher may filter vs a producer doesn't

Maybe we could unify them under a `Processor` interface and compose the behaviours.

Reply:
a producer takes info from the outside and normalizes it
an enricher takes info from the inside and updates it with extra info

if we merge them, how are we going to differentiate what runs when?

Reply:
The producer still reads the parsed vuln findings, does something with them and outputs them somewhere (the enriched does a similar thing).

Could the different behaviours between them be just implementation details rather than being different components at all?